compiler/util/files.cpp

```diff
@@ -171,13 +192,16 @@ void restoreLibraryAndIncludeInfo() {
   INT_ASSERT(
       fDriverPhaseTwo &&
       "should only be restoring library and include info in driver phase two");
-  assert(libDirs.empty() && libFiles.empty() && incDirs.empty() &&
-         "tried to restore library and include info from disk, but it was "
-         "already present in memory");
 
-  restoreDriverTmp(libDirsFilename, &addLibPath);
-  restoreDriverTmp(libFilesFilename, &addLibFile);
-  restoreDriverTmp(incDirsFilename, &addIncInfo);
+  restoreDriverTmp(libDirsFilename, [](const char* filename) {
+    addLibPath(filename, /* fromCmdLine */ false);
+  });
+  restoreDriverTmp(libFilesFilename, [](const char* filename) {
+    addLibFile(filename, /* fromCmdLine */ false);
+  });
+  restoreDriverTmp(incDirsFilename, [](const char* filename) {
+    addIncInfo(filename, /* fromCmdLine */ false);
+  });
```

Comment:
Won't setting `fromCmdLine` to `false` here means that restoring from the file will also write to the file -- based on the following code:

```C++
  if (fDriverPhaseOne && !fromCmdLine) {
    saveDriverTmp(incDirsFilename, incDir);
  }
```

Perhaps it makes sense to just make that boolean "saveToDisk", true by default, and overwrite it to false in CLI invocations?

Reply:
It won't because these calls are only reachable from driver phase two (`assert`ed right before) and that `if` prevents writing outside of phase one.
Nevertheless, do you think changing to `saveToDisk` like you described would make things clearer? My thinking was to limit how much code is aware of or references driver functionality, including saving to tmp files.

Reply:
> It won't because these calls are only reachable from driver phase two (asserted right before) and that if prevents writing outside of phase one.

Ah, obviously. Guess I'm still waking up.

I think keeping it as-is is more declarative, and since your logic handles it just fine, I see no reason to change it to `saveToDisk` after all.